api/src/pcapi/core/finance/api.py

```diff
@@ -1494,12 +1500,65 @@ def _mark_as_processed(pricings: sqla_orm.Query) -> None:
                     )
                     continue
 
-                total = pricings.with_entities(sqla.func.sum(models.Pricing.amount)).scalar()
-                if not total:
+                total = pricings.with_entities(sqla.func.sum(models.Pricing.amount)).scalar() or 0
+
+                all_current_incidents = (
```

Comment:
Cette variable n'est utilisée que dans certains cas précis (dans un des `if` plus bas). On pourrait déplacer sa définition pour éviter une requête SQL inutile.

Reply:
Je ne comprends pas ce commentaire dès lors qu'il n'y a plus de `if not total`. Cela ressemble plus à ça : 
```python
total = pricings.with_entities(sqla.func.sum(models.Pricing.amount)).scalar() or 0

                all_current_incidents = (
                    models.FinanceIncident.query.join(models.FinanceIncident.booking_finance_incidents)
                    .join(models.BookingFinanceIncident.finance_events)
                    .join(models.FinanceEvent.pricings)
                    .outerjoin(models.Pricing.cashflows)
                    .filter(
                        models.FinanceIncident.venueId.in_(venue_ids),
                        models.FinanceIncident.status == models.IncidentStatus.VALIDATED,
                        models.Cashflow.id.is_(None),  # exclude incidents that already have a cashflow
                        models.Pricing.status == models.PricingStatus.VALIDATED,
                        models.Pricing.valueDate < batch.cutoff,
                    )
                    .all()
                )

                # The total is positive if the pro owes us more than we do.
                if total >= 0:
                    last_cashflow_age = 0

                    if total > 0:
```

Reply:
`all_current_incidents` n'est utilisé que si `total > 0`. Rien ne sert de le calculer dans tous les cas.